rescript/screenseq.py

```diff
@@ -35,10 +34,9 @@ def _filter_seq(seq_obj, num_degenerates, homopolymer_length):
 
 
 def cull_seqs(sequences: DNAIterator, num_degenerates: int = 5,
-              homopolymer_length: int = 8) -> DNAFASTAFormat:
+              homopolymer_length: int = 8, n_jobs: int = 1) -> DNAFASTAFormat:
     result = DNAFASTAFormat()
 
-    n_jobs = cpu_count()
     parallel = Parallel(n_jobs=n_jobs, backend='loky')
     seqs = parallel(
```

Comment:
I have a naive question. I am not all that familiar with how `Parallel` works... Does this keep all `seq` data in memory prior to writing it out later? If so, we should avoid this. Users often run into memory issues, which is why we try to read/write the the data as it is being processed. Can this be adjusted so that if using 1 CPU, that it will read and write the sequences as it is culling? Or something similar? I think @BenKaehler, has been able to write out separate chunked files using while using multiple jobs, then merged those files together at the end.  At least that is what I think was done. Perhaps I am mis-understanding...

Reply:
That's a very good point and yes, that's how it works - everything is stored in memory. I tried before some other approaches but everything was just too slow. I also tried out your suggestion - it works, in principle, but the fact of constantly opening and writing to new files slows it down significantly... (if I do it on a per-sequence basis). As a last approach, I tried writing to the final file directly from the worker processes using a lock - that works pretty nicely and is as fast as the version before. The only drawback here is that the order of sequences in the new file won't necessarily be preserved - do you think this is a problem? In any case, feel free to check out the update in the new commits. Thanks!

Reply:
Thank you so much @misialq! I do not think order will be a problem for the intended use cases. We'll be culling sequences anyway. If it will be a problem for some users, then we can simply make note of this in the user help text. If they would like to keep it in order, then they can use 1 CPU. Or... we could do something insane and provide access to both approaches as two different actions. One that is for low-memory users, and another as you initially had it for those with ample memory. Though that might make things more complicated... We can always return to this if it is a problem.  I'll take a look at the new commit now. :-)